packages/discord.js/src/managers/ApplicationCommandPermissionsManager.js

```diff
@@ -300,43 +313,67 @@ class ApplicationCommandPermissionsManager extends BaseManager {
       if (error.code !== RESTJSONErrorCodes.UnknownApplicationCommandPermissions) throw error;
     }
 
-    const permissions = existing.filter(perm => !resolvedIds.includes(perm.id));
+    const permissions = existing.filter(perm => {
+      switch (perm.type) {
+        case ApplicationCommandPermissionType.Role:
+          return !resolvedRoleIds.includes(perm.id);
+        case ApplicationCommandPermissionType.User:
+          return !resolvedUserIds.includes(perm.id);
+        case ApplicationCommandPermissionType.Channel:
+          return !resolvedChannelIds.includes(perm.id);
+      }
+      return true;
+    });
 
-    return this.set({ guild: guildId, command: commandId, permissions });
+    return this.set({ guild: guildId, command: commandId, permissions, token });
   }
 
   /**
    * Options used to check the existence of permissions on a command
    * <warn>The `command` parameter is not optional when the managers `commandId` is `null`</warn>
    * @typedef {BaseApplicationCommandPermissionsOptions} HasApplicationCommandPermissionsOptions
-   * @property {UserResolvable|RoleResolvable} permissionId The user or role to check if a permission exists for
+   * @property {ApplicationCommandPermissionIdResolvable} permissionId The entity to check if a permission exists for
    * on this command.
+   * @property {ApplicationCommandPermissionType} [permissionType] Check for a specific type of permission
    */
 
   /**
-   * Check whether a permission exists for a user or role
-   * @param {AddApplicationCommandPermissionsOptions} options Options used to check permissions
+   * Check whether a permission exists for a user, role, or channel
+   * @param {HasApplicationCommandPermissionsOptions} options Options used to check permissions
    * @returns {Promise<boolean>}
    * @example
    * // Check whether a user has permission to use a command
    * guild.commands.permissions.has({ command: '123456789012345678', permissionId: '876543210123456789' })
    *  .then(console.log)
    *  .catch(console.error);
    */
-  async has({ guild, command, permissionId }) {
+  async has({ guild, command, permissionId, permissionType }) {
```

Comment:
😢 Do we want to check this when resolving the id as well? (Or just only accept ids like above?)

Reply:
I mean permissionType here doesn't really make sense, since you pass in an id anyways... I'd assume you know if you're passing in an id of a user vs one of a channel... CC @discordjs/the-big-4 tho, see what they think

Reply:
This explicitly exists for the edge case where there is a channel id that is the same as the guild id, which is the same as the everyone role id. The function needs to know whether you're checking for a channel or role overwrite